### ISSUE TITLE
Inductor 1D strided stores lower to scalar scatter instead of 2D block writes

### ISSUE THREAD

**Issue by etiotto** [intel/intel-xpu-backend-for-triton#6532]
## Problem

PyTorch Inductor generates Triton kernels that store into slices of concatenated tensors using 1D tensor-of-pointers. The addresses follow the pattern `base + (idx % W) + (idx // W) * S` — a row-major 2D tile flattened into 1D. Because the store is rank-1, `MaterializeBlockPointer` skips it and the store lowers to scalar scatter writes (`PredicatedStore`).

## Reproducer

```python
@triton.jit
def kernel(in_ptr, out_ptr, xnumel, W: tl.constexpr, S: tl.constexpr, XBLOCK: tl.constexpr):
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.arange(0, XBLOCK)
    xmask = tl.full([XBLOCK], True, tl.int1)
    val = tl.load(in_ptr + xindex, xmask)
    out_addr = (xindex % W) + (xindex // W) * S
    tl.store(out_ptr + out_addr, val, xmask)
```

With `W=32, S=96, f16, XBLOCK=1024, num_warps=4` — LLIR shows `PredicatedStore` instead of `LSC2DBlockWrite`.

**Comment by etiotto** [comment]
Here is a microbenchmark to evaluate the performance of the store instructions.
The kernel computes out_addr = (xindex % W) + (xindex // W) * S — that's a 2D row-major address pattern (col + row * stride) encoded in a 1D index. The tl.full([XBLOCK], True, tl.int1) mask means no elements are conditionally skipped, so the entire tile can be written as one 2D block store instead of per-element scatter writes.

```
"""Benchmark 1D strided stores at scale — no mask (splat true).

Matches the Inductor pattern where xnumel is a multiple of XBLOCK,
so the mask is always true and the reshape can trigger.
"""
import triton
import triton.language as tl
import torch
import numpy as np


@triton.jit
def strided_store_nomask(
    in_ptr, out_ptr,
    xnumel,
    W: tl.constexpr, S: tl.constexpr,
    XBLOCK: tl.constexpr,
):
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.arange(0, XBLOCK)
    xmask = tl.full([XBLOCK], True, tl.int1)
    val = tl.load(in_ptr + xindex, xmask).to(tl.float32)
    result = val * 0.5  # simple compute to avoid being optimized away
    result = result.to(tl.float16)
    out_addr = (xindex % W) + (xindex // W) * S
    tl.store(out_ptr + out_addr, result, xmask)


def bench_config(W, S, dtype, num_rows, xblock, num_warps, label):
    xnumel = W * num_rows
    assert xnumel % xblock == 0, f"xnumel={xnumel} not divisible by XBLOCK={xblock}"
    in_tensor = torch.randn(xnumel, device='xpu', dtype=dtype)
    out_tensor = torch.zeros(num_rows * S, device='xpu', dtype=dtype)

    grid = (xnumel // xblock,)

    # Warmup
    for _ in range(10):
        strided_store_nomask[grid](
            in_tensor, out_tensor, xnumel,
            W=W, S=S, XBLOCK=xblock, num_warps=num_warps,
        )
    torch.xpu.synchronize()

    # Benchmark
    times = []
    for _ in range(100):
        start = torch.xpu.Event(enable_timing=True)
        end = torch.xpu.Event(enable_timing=True)
        start.record()
        strided_store_nomask[grid](
            in_tensor, out_tensor, xnumel,
            W=W, S=S, XBLOCK=xblock, num_warps=num_warps,
        )
        end.record()
        torch.xpu.synchronize()
        times.append(start.elapsed_time(end))

    ms = np.median(times)
    elem_size = in_tensor.element_size()
    num_bytes = xnumel * elem_size * 2
    gb_per_s = (num_bytes / 1e9) / (ms / 1e3)
    print(f"  {label:55s}  {ms:.4f}ms  {gb_per_s:8.2f} GB/s")
    return ms, gb_per_s


def main():
    print("=" * 90)
    print("Benchmark: 1D strided stores — no mask (Inductor pattern)")
    print("=" * 90)

    configs = [
        # (W, S, dtype, num_rows, xblock, num_warps, label)
        # xnumel must be divisible by xblock

        # W=32 variants (original Inductor case)
        (32, 96,  torch.float16, 307200, 1024, 4, "W=32 S=96  f16 ~10M (original pattern)"),
        (32, 128, torch.float16, 307200, 1024, 4, "W=32 S=128 f16 ~10M"),
        (32, 192, torch.float16, 307200, 1024, 4, "W=32 S=192 f16 ~10M"),

        # Narrow width — partial cache lines
        (16, 48,  torch.float16, 640000, 1024, 4, "W=16 S=48  f16 10M (half cache line)"),
        (16, 96,  torch.float16, 640000, 1024, 4, "W=16 S=96  f16 10M"),

        # f32 (larger element)
        (32, 96,  torch.float32, 307200, 1024, 4, "W=32 S=96  f32 ~10M"),
        (32, 192, torch.float32, 307200, 1024, 4, "W=32 S=192 f32 ~10M"),

        # Very large
        (32, 96,  torch.float16, 3072000, 1024, 4, "W=32 S=96  f16 ~100M"),
        (32, 192, torch.float16, 3072000, 1024, 4, "W=32 S=192 f16 ~100M"),
    ]

    for W, S, dtype, num_rows, xblock, num_warps, label in configs:
        try:
            bench_config(W, S, dtype, num_rows, xblock, num_warps, label)
        except Exception as e:
            print(f"  {label:55s}  FAILED: {e}")


if __name__ == '__main__':
    main()
```

**Comment by etiotto** [comment]
For the benchmark code at https://github.com/intel/intel-xpu-backend-for-triton/issues/6532#issuecomment-4164631872, the performance with latest main is:

| Config | Bandwidth (GB/s) |
|--------|-----------------|
| W=32 S=96 f16 10M | 350.46 |
| W=32 S=128 f16 10M | 459.15 |
| W=32 S=192 f16 10M | 349.84 |
| W=16 S=48 f16 10M | 320.80 |
| W=16 S=96 f16 10M | 305.85 |
| W=32 S=96 f32 10M | 753.87 |
| W=32 S=192 f32 10M | 426.85 |
| W=32 S=96 f16 100M | 616.21 |
| W=32 S=192 f16 100M | 605.39 |

With the code in PR https://github.com/intel/intel-xpu-backend-for-triton/pull/6531 the performance is:

| Config | Bandwidth (GB/s) |
|--------|-----------------|
| W=32 S=96 f16 10M | 818.52 |
| W=32 S=128 f16 10M | 833.79 |
| W=32 S=192 f16 10M | 808.42 |
| W=16 S=48 f16 10M | 685.87 |
| W=16 S=96 f16 10M | 724.19 |
| W=32 S=96 f32 10M | 1330.23 |
| W=32 S=192 f32 10M | 580.31 |
| W=32 S=96 f16 100M | 710.80 |
| W=32 S=192 f16 100M | 630.28 |

The feature improves performance by an average of 81% (PVC 1100).
